### DISCUSSION TITLE
官方是否有计划将 事件的添加方式由  onxxx 改为 addEventListener('xx', ...)，以避免像 onwheel/touchstart/touchmove 在浏览器中报 "passive" 警告？

### DISCUSSION THREAD

**Discussion by puxiao** [antvis/G6#5026] [Show and tell]
目前当运行 G6 时谷歌浏览器会收到一大堆关于 onwheel/touchstart/touchmove... 相关的 "passive" 警告信息。

> [Violation] Added non-passive event listener to a scroll-blocking <some> event. Consider marking event handler as 'passive' to make the page more responsive.
>
>  Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive.
>
> .....

<br>

我查看了一下源码，发现对于各种事件的添加是通过 graph.emit( ... )  方式来进行的，至于 .emit() 又由 "EventEmitter" 来管理维护的。

> import EventEmitter from '@antv/event-emitter'

还有一部分事件竟然用的是 `import { addEventListener } from '@antv/dom-util';` ，而依赖的 @antv/dom-util 是比较旧的 2.0.1 。
最新版 3.3.4 好像已经移除了 addEventListener

<br>

我们知道像下面的代码就会收到 "passive" 相关警告：

```
const myDiv = document.getElementById('myDiv')

const handleWheel = (e: WheelEvent) => {
    console.log(e)
    e.preventDefault()
}

myDiv.onwheel = handleWheel
```

<br>

将 onwheel 改为 addEventListener 并且增加 "{ passive: false }" 就可以消除上述警告信息

```diff
- myDiv.onwheel = handleWheel
+ myDiv.addEventListener( 'wheel', handleWheel, { passive: false } )
```

<br>

官方是否有计划统一修改呢？

<br>

**Comment by Aarebecca** [comment]
感谢你的提示，我们后续会处理这个问题
